src/indieweb_utils/posts/discovery.py

```diff
@@ -0,0 +1,270 @@
+import re
+
+import mf2py
+import requests
+from bs4 import BeautifulSoup
+
+
+def discover_original_post(posse_permalink: str):
+    """
+    Find the original version of a post per the Original Post Discovery algorithm.
+
+    refs: https://indieweb.org/original-post-discovery#Algorithm
```

Comment:
Added a link to the source Algorithm spec so we don't need to look it up each time we want to reference it.